ec2 facts module - registers to ansible_facts key

silviud opened the pull request:
Hi,

I discovered a bug into the previous version (I closed the request). It had to do with the fact that the module was not registering the ec2 facts to ansible_facts key.

Tested into a playbook as well 

```

---
- hosts: ec2_host
  gather_facts: yes
  tasks:
    - name: ec2 facts
      action:  ec2_facts
    - name: command
      action: command echo "ec2 facts tests"
      when_string: $ansible_ec2_profile == "default-paravirtual"
```

Sorry for inconvenience.

silviud committed:
ec2 module - registers to ansible_facts key

mpdehaan commented:
Looks pretty good.  In queue for further review.

Some initial comments:

I would change the documentation string "can be add it on"  to something like "this may be added later"

This module should not need to return 'changed' since it's not making any side effects, so that should also be removed.

(In this case, you can update this pull request without resubmitting it)

Thanks!

silviud committed:
ec2 facts moduled - updated

lwade commented:
Tested and it works fine, thanks.  I'll put this in the comments rather than a diff:

Sorry if this seems pedantic but I would change all instances of "metadata servers" to "the metadata service".  I'd also change any instance of "url to uri", thus both descriptions and variables should be like:

ec2_metadata_uri  

This module works fine in both AWS and Eucalyptus due to the compat between them, so perhaps mention this in the module comments and code.

lwade commented:
May I also suggest that you just add extra uri's for:

block-device-mapping/
network/
placement/

for now, following the same method you demonstrate in this module.  Sure, perhaps a solution for running through the entire tree for _everything_ would be nice but in the meantime this quick method would do the job of returning the information.

silviud committed:
ec2 facts module - updated  as per comments

silviud commented:
Hello,

I incorporated the changes that lwade suggested and updated the docstring as well the variable names,
The implementation changed a bit in order to walk the uri tree.

thanks!

ps: I have three commits in here (instead of one) but I'm not sure how I can rebase/squash if the history is present on the git server already.

mpdehaan commented:
thanks -- in queue for review and such.

Most likely this weekend -- planning a lot of combing through pull requests
and sharpening up 1.0 docs.

On Thu, Jan 24, 2013 at 7:19 PM, silviud notifications@github.com wrote:

> Hello,
> 
> I incorporated the changes that lwade suggested and updated the docstring
> as well the variable names,
> The implementation changed a bit in order to walk the uri tree.
> 
> thanks!
> 
> ps: I have three commits in here (instead of one) but I'm not sure how I
> can rebase/squash if the history is present on the git server already.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ansible/ansible/pull/1891#issuecomment-12681636.

lwade commented:
Excellent, thanks silviud.  

Tested this also, works a treat :)

mpdehaan commented:
Merged, thanks!
